### PR TITLE
A few package used in HEP including ROOT6

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+from spack import *
+
+class Geant4(Package):
+    """Geant4 is a toolkit for the simulation of the passage of particles through matter. Its areas of application include high energy, nuclear and accelerator physics, as well as studies in medical and space science."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/geant4/package.py]: Line is too long. Wrap around so it is less than 79 characters.

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    depends_on("clhep@2.3.1.1~cxx11+cxx14", when="@10.02.p01")
+    depends_on("clhep@2.2.0.4~cxx11+cxx14", when="@10.01.p03")
+    depends_on("expat")
+    depends_on("zlib")
+    depends_on("xerces-c")
+    depends_on("qt@4.8:", when="+qt")
+
+    def install(self, spec, prefix):
+        cmake_args = list(std_cmake_args)
+        cmake_args.append('-DXERCESC_ROOT_DIR:STRING=%s'%spec['xerces-c'].prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/geant4/package.py]: Probably going to need spaces around that percent symbol. Flake8 will complain.

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        if '+qt' in spec:
+            cmake_args.append('-DGEANT4_USE_QT=ON')
+
+        build_directory = join_path(self.stage.path, 'spack-build')
+        source_directory = self.stage.source_path
+
+        with working_dir(build_directory, create=True):
+            cmake(source_directory, *cmake_args)
+            make()
+            make("install")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/geant4/package.py]: You should just be able to use:
```python
with working_dir('spack-build', create=True)
    cmake('..', *cmake_args)

    make()
    make('install')
```

File: var/spack/repos/builtin/packages/gmake/package.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+#     spack install gmake
+#
+# You can edit this file again by typing:
+#
+#     spack edit gmake
+#
+# See the Spack documentation for more information on packaging.
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gmake/package.py]: Remove template.

File: var/spack/repos/builtin/packages/gmake/package.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+from spack import *
+
+
+class Gmake(Package):
+    """FIXME: Put a proper description of your package here."""
+
+    # FIXME: Add a proper url for your package's homepage here.
+    homepage = "http://www.example.com"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gmake/package.py]: Need docstring, homepage

File: var/spack/repos/builtin/packages/gmake/package.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    url      = "ftp://ftp.gnu.org/gnu/make/make-4.0.tar.gz"
+
+    version('4.0', 'b5e558f981326d9ca1bfdb841640721a')
+
+    # FIXME: Add dependencies if required.
+    # depends_on('foo')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gmake/package.py]: Remove if no dependencies.

File: var/spack/repos/builtin/packages/gmake/package.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+    def install(self, spec, prefix):
+        # FIXME: Modify the configure line to suit your build system here.
+        configure('--prefix={0}'.format(prefix))
+
+        # FIXME: Add logic to build and install here.
+        make()
+        make('install')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gmake/package.py]: Remove FIXME comments if everything works.

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+##############################################################################
+
+from spack import *
+
+
+class Hepmc(Package):
+    """FIXME: Put a proper description of your package here."""
+
+    # FIXME: Add a proper url for your package's homepage here.
+    homepage = "http://www.example.com"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]: Docstring, homepage.

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    url      = "http://lcgapp.cern.ch/project/simu/HepMC/download/HepMC-2.06.07.tar.gz"
+
+    version('2.06.09', 'c47627ced4255b40e731b8666848b087')
+    version('2.06.08', 'a2e889114cafc4f60742029d69abd907')
+    version('2.06.07', '11d7035dccb0650b331f51520c6172e7')
+    version('2.06.06', '102e5503537a3ecd6ea6f466aa5bc4ae')
+    version('2.06.05', '2a4a2a945adf26474b8bdccf4f881d9c')
+
+    # FIXME: Add additional dependencies if required.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]: Remove FIXME

File: var/spack/repos/builtin/packages/heppdt/package.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+from spack import *
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/heppdt/package.py]: Missing license.

File: var/spack/repos/builtin/packages/heppdt/package.py [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+class Heppdt(Package):
+    """ HEPPDT ."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/heppdt/package.py]: Why not a more informative docstring, like:

> The HepPID library contains translation methods for particle ID's
to and from various Monte Carlo generators and the PDG standard
numbering scheme.  We realize that the generators adhere closely
to the standard, but there are occasional differences.

This comes from the package's README

File: var/spack/repos/builtin/packages/root/package.py [hunk 12]
```diff
@@ -0,0 +1,70 @@
+from spack import *
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/root/package.py]: Missing license

File: var/spack/repos/builtin/packages/root/package.py [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    depends_on("libxml2+python")
+    depends_on("jpeg")
+    if sys.platform != 'darwin': 
+        depends_on("libpng")
+        depends_on("openssl")
+        depends_on("freetype")
+
+    def install(self, spec, prefix):
+
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/root/package.py]: Flake8 won't like all of these empty lines.

File: var/spack/repos/builtin/packages/xrootd/package.py [hunk 14]
```diff
@@ -0,0 +1,26 @@
+from spack import *
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xrootd/package.py]: Missing license.

File: var/spack/repos/builtin/packages/xrootd/package.py [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+
+class Xrootd(Package):
+    """The XROOTD project aims at giving high performance, scalable fault tolerant access to data repositories of many kinds."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xrootd/package.py]: Line too long.

File: var/spack/repos/builtin/packages/geant4/package.py (hunk 3 above)
Comment (gartung) [inline var/spack/repos/builtin/packages/geant4/package.py]: Need to do an out of source tree build.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/geant4/package.py]: Isn't that what this does? It creates a new directory called `spack-build` and builds from there.

Comment (gartung) [inline var/spack/repos/builtin/packages/geant4/package.py]: I don't know. The original author wanted it that way so I left it alone.

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+class Geant4(Package):
+    """Geant4 is a toolkit for the simulation of the passage of particles through matter.\
+        Its areas of application include high energy, nuclear and accelerator physics, \
+        as well as studies in medical and space science."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/geant4/package.py]: In Python, triple quotes signify a multiline comment. As such, the backslashes at the end of the lines aren't necessary. Instead, they create a single line comment that contains long stretches of whitespace. I believe that is what is causing the documentation tests to fail with messages like:

```
Warning, treated as error:
/home/travis/build/LLNL/spack/lib/spack/docs/package_list.rst:4770: WARNING: Block quote ends without a blank line; unexpected unindent.
```

**Comment by adamjstewart** [comment]
Can you address the remaining Flake8 errors? See the bottom of https://travis-ci.org/LLNL/spack/jobs/162039765 for details.

**Comment by gartung** [comment]
Fixed all flake8 errors except line too long for urls.

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Geant4(Package):
+    """Geant4 is a toolkit for the simulation of the passage of particles
+through matter. Its areas of application include high energy, nuclear
+and accelerator physics, as well as studies in medical and space science."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/geant4/package.py]: The start of these lines need to be indented 4 spaces.

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Hepmc(Package):
+    """The HepMC package is an object oriented, C++ event record for \
+       High Energy Physics Monte Carlo generators and simulation."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]: Need to remove backslash.

File: var/spack/repos/builtin/packages/heppdt/package.py [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+##############################################################################
+
+from spack import *
+
+
+class Heppdt(Package):
+    """ The HepPID library contains translation methods for particle ID's \
+        to and from various Monte Carlo generators and the PDG standard \
+        numbering scheme. We realize that the generators adhere closely \
+        to the standard, but there are occasional differences."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/heppdt/package.py]: Need to remove backslashes.

**Comment by gartung** [comment]
Pushed the doc changes. Not sure what is failing now.

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Geant4(Package):
+    """Geant4 is a toolkit for the simulation of the passage of particles
+    through matter. Its areas of application include high energy, nuclear
+    and accelerator physics, as well as studies in medical and space science."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/geant4/package.py]: Now that it's indented properly, this line is just one character too long. You can either move "science" onto the next line or simply move the triple quotes to the next line. By the way, if you want to run these tests locally, you can install flake8, add it to your PATH, and run `share/spack/qa/run-flake8-tests`.

File: var/spack/repos/builtin/packages/heppdt/package.py [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+##############################################################################
+
+from spack import *
+
+
+class Heppdt(Package):
+    """ The HepPID library contains translation methods for particle ID's
+        to and from various Monte Carlo generators and the PDG standard
+        numbering scheme. We realize that the generators adhere closely
+        to the standard, but there are occasional differences."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/heppdt/package.py]: I think this is the last thing that the documentation tests are unhappy about. When building the documentation, we run `spack package-list` to create [this beautiful page](http://spack.readthedocs.io/en/latest/package_list.html). `spack package-list` prints out an .rst file based on the docstrings of every package. But if the docstring is improperly formatted, the indentation in `package_list.rst` can be messed up. To fix this one, I think you need to remove the leading spaces so it looks like:
```python
"""The HepPID library contains translation methods for particle ID's
to and from various Monte Carlo generators and the PDG standard
numbering scheme. We realize that the generators adhere closely
to the standard, but there are occasional differences."""
```
By the way, if you want to run these tests locally, there's a `share/spack/qa/run-doc-tests` command that will try building the documentation and report any errors. You just need sphinx, graphviz, git, mercurial, and svn installed. All of them should be in Spack.

**Comment by gartung** [comment]
OK. Hopefully these changes work.

File: var/spack/repos/builtin/packages/heppdt/package.py [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+##############################################################################
+
+from spack import *
+
+
+class Heppdt(Package):
+    """ The HepPID library contains translation methods for particle ID's
+to and from various Monte Carlo generators and the PDG standard
+numbering scheme. We realize that the generators adhere closely
+to the standard, but there are occasional differences."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/heppdt/package.py]: To clarify, it needs to look like this:
```python
class Heppdt(Package):
    """The HepPID library contains translation methods for particle ID's
    to and from various Monte Carlo generators and the PDG standard
    numbering scheme. We realize that the generators adhere closely
    to the standard, but there are occasional differences."""
```

**Comment by tgamblin** [comment]
Squashed and merged with one last doc fix.

**Comment by tgamblin** [comment]
Thanks @gartung!

**Pull request merged**